tests/mpi/affine_constraints_get_view_01.cc

```diff
@@ -192,8 +192,14 @@ test()
     pressure_constraints.distribute(check_vector_2.block(1));
   }
 
-  // Now the assertion part: These vectors are the same:
-  Assert(check_vector_1 == check_vector_2, ExcInternalError());
+  // Finally check that these two vectors are the same. Note that
+  // when we compile with native optimizations we might have a slight
+  // difference in results:
+  {
+    TrilinosWrappers::MPI::BlockVector result = check_vector_1;
+    result -= check_vector_2;
+    Assert(result.l2_norm() < 1e-10, ExcInternalError());
```

Comment:
Can you make this a relative tolerance? I don't recall how I set up the vector entries, but relative should always be safe.